internal/controller/garbage_collector_runner_test.go

```diff
@@ -0,0 +1,159 @@
+package controller
+
+import (
+	"time"
+
+	mantlev1 "github.com/cybozu-go/mantle/api/v1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("garbage collector", func() {
+	var err error
+
+	Context("isMantleRestoreAlreadyDeleted", func() {
+		doTest :=
+			func(
+				ctx SpecContext,
+				existMR bool,
+				pvModifier func(pv *corev1.PersistentVolume),
+				expectError, expectDeleted bool,
+			) {
+				runner := NewGarbageCollectorRunner(k8sClient, 1*time.Second, "rook-ceph")
+				ns := resMgr.CreateNamespace()
+
+				var mr mantlev1.MantleRestore
+				mr.SetName("mr")
+				mr.SetNamespace(ns)
+				mr.Spec.Backup = "dummy"
+				if existMR {
+					err = k8sClient.Create(ctx, &mr)
+					Expect(err).NotTo(HaveOccurred())
+					err = k8sClient.Get(ctx, types.NamespacedName{Name: mr.GetName(), Namespace: mr.GetNamespace()}, &mr)
+					Expect(err).NotTo(HaveOccurred())
+				}
+
+				var pv corev1.PersistentVolume
+				pv.SetAnnotations(map[string]string{
+					PVAnnotationRestoredBy:          string(mr.GetUID()),
+					PVAnnotationRestoredByName:      "mr",
+					PVAnnotationRestoredByNamespace: ns,
+				})
+				pv.Spec.CSI = &corev1.CSIPersistentVolumeSource{
+					VolumeAttributes: map[string]string{"clusterID": "rook-ceph"},
+				}
+				if pvModifier != nil {
+					pvModifier(&pv)
+				}
+
+				deleted, err := runner.isMantleRestoreAlreadyDeleted(ctx, &pv)
+				if expectError {
+					Expect(err).To(HaveOccurred())
+				} else {
+					Expect(err).NotTo(HaveOccurred())
+				}
+				Expect(deleted).To(Equal(expectDeleted))
+			}
+
+		DescribeTable("isMantleRestoreAlreadyDeleted",
+			doTest,
+			Entry("MantleRestore exists", true, nil, false, false),
+			Entry("MantleRestore does NOT exist", false, nil, false, true),
+			Entry(
+				"MantleRestore exists, but the PV's annotation has unexpected MR's UID",
+				true,
+				func(pv *corev1.PersistentVolume) {
+					pv.Annotations[PVAnnotationRestoredBy] = "unexpected-uid"
+				},
+				false, true,
+			),
+			Entry(
+				"PV annotation of restored-by missing",
+				true,
+				func(pv *corev1.PersistentVolume) {
+					delete(pv.Annotations, PVAnnotationRestoredBy)
+				},
+				true, false,
+			),
+			Entry(
+				"PV annotation of restored-by-name missing",
+				true,
+				func(pv *corev1.PersistentVolume) {
+					delete(pv.Annotations, PVAnnotationRestoredByName)
+				},
+				true, false,
+			),
+			Entry(
+				"PV annotation of restored-by-namespace missing",
+				true,
+				func(pv *corev1.PersistentVolume) {
+					delete(pv.Annotations, PVAnnotationRestoredByNamespace)
+				},
+				true, false,
+			),
```

Comment:
Could you add one more test for the check of clusterID?

```suggestion
			),
			Entry(
				"cluster ID does not match",
				true,
				func(pv *corev1.PersistentVolume) {
					pv.Spec.CSI = &corev1.CSIPersistentVolumeSource{
						VolumeAttributes: map[string]string{"clusterID": "different-cluster"},
				}				},
				false, false,
			),
```

Reply:
Fixed in https://github.com/cybozu-go/mantle/pull/71/commits/1d60ebc98f03f7ac21ccc914788669e3a9cb1005.